src/features/events/components/EventSelectionCheckBox.tsx

```diff
@@ -0,0 +1,84 @@
+import { Box, Checkbox } from '@mui/material';
+import { useSelector, useStore } from 'react-redux';
+
+import { eventsSelected } from '../store';
+import { RootState } from 'core/store';
+import { ZetkinEvent } from 'utils/types/zetkin';
+
+interface EventSelectionCheckBoxProps {
+  eventList: ZetkinEvent[];
+  multiSelect?: boolean;
+}
+const EventSelectionCheckBox = ({
```

Comment:
I would personally have chosen another path than this, but I think this is a fine solution, to have a component that handles this. I would just change some minor things if we do continue down this path.